ui/app/js/controllers/user_edit_controller.js

```diff
@@ -281,10 +299,9 @@ function ($scope, $modalInstance, CSRF, Permissions, user, users) {
 
 
   $scope.cancel = function () {
-      if (!$scope.user.permissions.length) {
-        $scope.users.pop($scope.user.username);
-      }
-      $modalInstance.dismiss('cancel');
-
+    if (!$scope.is_edit){
+      $scope.users.push($scope.user);
+    }
+    $modalInstance.dismiss('cancel');
```

Comment:
This causes a user with no permissions to be added to the Users list, which is a bit confusing. Did you intend this?

Reply:
No.. I didn't notice the user was being added even without permission. Will fix it.